### PR TITLE
Start using helpers from k8s.io/component-helpers

### PR DESCRIPTION
**Pull request by damemi**
With the new `k8s.io/component-helpers` repo, we can start replacing many of the helpers that were duplicated in https://github.com/kubernetes-sigs/descheduler/commit/431597dd43667b681a2fd3c4f782bd6ef5e41bed with their new external equivalents. 

This function will depend on the PR https://github.com/kubernetes/kubernetes/pull/95531 to make it available in the new repo

### CODE REVIEW

**Comment by damemi** [comment]
/hold
for upstream PR

**Review by damemi — COMMENTED** [review]
/hold cancel
/cc @seanmalloy @ingvagabund 
This sets us up to start importing more helpers and remove dupe code 🎉

File: go.mod [hunk 1]
```diff
@@ -5,11 +5,12 @@ go 1.15
...
-	k8s.io/api v0.19.2
-	k8s.io/apimachinery v0.19.2
+	k8s.io/api v0.20.0-beta.0
+	k8s.io/apimachinery v0.20.0-beta.0
 	k8s.io/apiserver v0.19.0
-	k8s.io/client-go v0.19.2
+	k8s.io/client-go v0.20.0-beta.0
 	k8s.io/code-generator v0.19.0
 	k8s.io/component-base v0.19.2
+	k8s.io/component-helpers v0.20.0-alpha.2.0.20201030045251-0fa5e458da80
```
Comment (ingvagabund) [inline go.mod:14]: Until there's the first release of k8s.io/component-helpers, there are no stability guarantees of the API in the right sense even though the stability intention is strongly enforced in  `k8s.io/component-helpers` PR reviews. Given the descheduler is the first repo outside of kubernetes org to import the new staging repo and for the stability assumption mentioned above, we can use the descheduler repo as a guinea pig until the first release is out.

**Review by ingvagabund — APPROVED** [review]
1. **Are there any stability guarantees of the imported helper function?** Stronly enforced during PR reviews
1. **Any blockers for importing k8s.io/component-helpers in the descheduler repository?** The repository is a staging repository, thus pulling only other staging repositories, not bringing anything from outside. The first release is aligned with 1.20 at which point descheduler will no longer rely on a non-release revision. In order to provide k8s stability guarantees of the descheduler itself, all k8s dependency versions need to align with the same k8s release. The next descheduler release is planned for 1.20 at which point all k8s dependencies (including k8s.io/component-helpers) will get pinned to 1.20.
1. **How using of k8s.io/component-helpers repo is going to affect this repository?** Removing and avoiding duplicates

In future, the descheduler will depend more on `k8s.io/component-helpers` once the scheduling framework is fully migrated under `k8s.io/kube-scheduler`. This is a necessary prerequisite to achieving that goal.

**Pull request merged**